Realm/RLMRealmConfiguration+Sync.mm

```diff
@@ -53,6 +53,8 @@ - (void)setSyncConfiguration:(RLMSyncConfiguration *)syncConfiguration {
 
     if (syncConfiguration.customFileURL) {
         self.config.path = syncConfiguration.customFileURL.path.UTF8String;
+    } else if (self.config.sync_config->flx_sync_requested) {
+        self.config.path = [user pathForPartitionValue:"default"];
```

Comment:
It would be good to have a descriptive constant here instead of `"default"`

Reply:
I changed this to use the path obtained from core for flexible sync